mongo/cursor.go

```diff
@@ -185,11 +185,20 @@ func (c *Cursor) Close(ctx context.Context) error {
 // This method requires driver version >= 1.1.0.
 func (c *Cursor) All(ctx context.Context, results interface{}) error {
 	resultsVal := reflect.ValueOf(results)
+
 	if resultsVal.Kind() != reflect.Ptr {
 		return errors.New("results argument must be a pointer to a slice")
 	}
 
 	sliceVal := resultsVal.Elem()
+	if sliceVal.Kind() == reflect.Interface {
+		sliceVal = sliceVal.Elem()
+	}
+
+	if sliceVal.Kind() != reflect.Slice {
+		return errors.New("results argument must be a slice address")
```

Comment:
This error and the one on line 190 seem really similar and it isn't clear to a user what failed if they hit one of them. Can you think of a way to include what we expected and what the user actually gave? For example, for this one, I think an error like "results argument must be a pointer to a slice, but was a pointer to integer" is much clearer.

Reply:
good idea. I've updated the error messages